test/teos/unit/test_responder.py

```diff
@@ -28,7 +28,7 @@
 @pytest.fixture(scope="module")
 def responder(db_manager, gatekeeper, carrier, block_processor):
     responder = Responder(db_manager, gatekeeper, carrier, block_processor)
-    chain_monitor = ChainMonitor(Queue(), responder.block_queue, block_processor, bitcoind_feed_params)
+    chain_monitor = ChainMonitor([Queue(), responder.block_queue], block_processor, bitcoind_feed_params)
```

Comment:
The fixture is not being activated. This is OK since the do_watch method is the only one that requires chain monitoring. However, this made me realise that, most likely, all the other tests could use the fixture.

Mind giving it a try? It may imply slightly modifying some of the tests (like monkey patching some of the state)

Reply:
PS: With monkey patching it may even work with only the fixture, since it does for the Watcher.

Reply:
Hopefully done in d124cc07560c6b209a068dcf79e44a813c5cc614.

Not sure if this is what you meant, but I changed the `responder` fixture to be per-test (so it's a clean responder every time) and added setup and teardown, and changed all the tests to use the fixture instead of creating their responder (slight difference to how tests were done before: awaking the responder and activating the chain_monitor is now part of the fixture).

Couldn't do it only in `test_do_watch` as the test seems to need more control on the creation and starting time.

Reply:
I may have to go over both Watchers and Responders test to simplify this further, so for now its OK.